src/main/java/org/elasticsearch/indices/query/IndicesQueriesModule.java

```diff
@@ -30,30 +30,21 @@
 
 public class IndicesQueriesModule extends AbstractModule {
 
-    private Set<Class<QueryParser>> queryParsersClasses = Sets.newHashSet();
-    private Set<QueryParser> queryParsers = Sets.newHashSet();
+    private Set<Class<? extends QueryParser>> queryParsersClasses = Sets.newHashSet();
 
-    public synchronized IndicesQueriesModule addQuery(Class<QueryParser> queryParser) {
+    public synchronized IndicesQueriesModule addQuery(Class<? extends QueryParser> queryParser) {
         queryParsersClasses.add(queryParser);
         return this;
     }
 
-    public synchronized IndicesQueriesModule addQuery(QueryParser queryParser) {
```

Comment:
I am conflicted on whether we should remove this or not.  I think the above `addQuery(Class<? extends QueryParser>)` is the way to go, while this one would force binding to a specific instance. Can restore it if there is some usecase for it, it's just that I don't like having too many ways to do the same thing.